patched-vscode/extensions/code-editor-idle-extension/src/extension.ts

```diff
@@ -0,0 +1,103 @@
+import * as vscode from "vscode";
+import * as fs from "fs";
+import * as path from "path";
+
+let idleFilePath: string
+let terminalActivityInterval: NodeJS.Timeout | undefined
+const CHECK_INTERVAL = 60000; // 60 seconds interval
+
+export function activate(context: vscode.ExtensionContext) {
+	initializeIdleFilePath();
+	registerEventListeners(context);
+	startMonitoringTerminalActivity();
+}
+
+export function deactivate() {
```

Comment:
should we also delete the file `.code-editor-last-active-timestamp` ?  
on a related note - if the file already exists when the app starts up, will there be any issue ?

Reply:
Yes, fs.writeFileSync will create the file if it does not already exist. If the file does exist, it will overwrite the existing content.